ios/app/brave_core_main.mm

```diff
@@ -89,6 +92,22 @@
 #include "ios/chrome/browser/credential_provider/model/credential_provider_util.h"
 #endif
 
+namespace brave {
+ProfileIOS* CreateMainProfileIOS() {
+  // Initialize and set the main browser state.
+  auto* localState = GetApplicationContext()->GetLocalState();
+  auto* profileManager = GetApplicationContext()->GetProfileManager();
+  std::string profileName = localState->GetString(prefs::kLastUsedProfile);
+  if (profileName.empty()) {
+    profileName = profileManager->ReserveNewProfileName();
+    localState->SetString(prefs::kLastUsedProfile, profileName);
+  }
+  DCHECK(!profileName.empty());
+  base::ScopedAllowBlocking allow_blocking;
+  return profileManager->CreateProfile(profileName);
```

Comment:
why aren't we using `CreateProfileAsync`?

Reply:
if for some reason it has to happen sync, can we just define `class ScopedAllowBlockingForProfile : public base::ScopedAllowBlocking {};` here and use that instead since it's already a friend?

Reply:
We need extensive refactors unfortunately to make it async so I had to use the synchronous version for the time being. We would eventually want shift it to use the async version. As for `ScopedAllowBlockingForProfile`, I didn't notice that, we can certainly swap that and get rid of the extra patch needed

Reply:
fixed in https://github.com/brave/brave-core/pull/26537/commits/4c5cb4726b37a08b9b9fa320658e2024871c562e